### PR TITLE
Remove signalr-client-aio dependency. Add it as a module

### PR DESCRIPTION
**Pull request by oscr**
After a lot of annoying connection errors, I finally got it running as a module.

I've added two example just to demonstrate and verify that it works. We might not want to keep them. But they are just there to show that the module in itself works. 

Tested on Ubuntu 22.04.1 LTS with Python 3.10.4

Fixes #245 

Example output:
```
python examples/example_fastf1_signalrclient.py 
selector_events      DEBUG 	Using selector: EpollSelector
client         INFO 	Starting FastF1 live timing client [v2.3.0]
connectionpool      DEBUG 	Starting new HTTPS connection (1): livetiming.formula1.com:443
connectionpool      DEBUG 	https://livetiming.formula1.com:443 "GET /signalr/negotiate?connectionData=%5B%7B%22name%22%3A+%22Streaming%22%7D%5D&clientProtocol=1.5 HTTP/1.1" 200 None
protocol      DEBUG 	= connection is CONNECTING
client        DEBUG 	> GET /signalr/connect?transport=webSockets&connectionToken=0ChwIbk5%2FvzqDbGRUhtGgR40YIxQeRlzAYaPZhIA7uVT1yICwF%2FXQdA2%2BxlziM0YHxUJN82VHUtEhWVU%2F0FwHL1BC3rImN0G9eBCFOSxX8nB5OipWVMkVC%2B2rNKjIm4x&connectionData=%5B%7B%22name%22%3A+%22Streaming%22%7D%5D&clientProtocol=1.5 HTTP/1.1
client        DEBUG 	> Host: livetiming.formula1.com
client        DEBUG 	> Upgrade: websocket
client        DEBUG 	> Connection: Upgrade
client        DEBUG 	> Sec-WebSocket-Key: EPqZ7hSC7iP0k7xdc403nw==
client        DEBUG 	> Sec-WebSocket-Version: 13
client        DEBUG 	> Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
client        DEBUG 	> User-agent: BestHTTP
client        DEBUG 	> Accept-Encoding: gzip, identity
client        DEBUG 	> Connection: keep-alive, Upgrade
client        DEBUG 	> Cookie: GCLB=CIzU08y17LPI6gE
client        DEBUG 	< HTTP/1.1 101 Switching Protocols
client        DEBUG 	< Upgrade: websocket
client        DEBUG 	< Server: Microsoft-HTTPAPI/2.0
client        DEBUG 	< X-server: streamrepeater-live-0chf
client        DEBUG 	< X-Content-Type-Options: nosniff
client        DEBUG 	< Connection: Upgrade
client        DEBUG 	< Sec-WebSocket-Accept: fjpvcisJ57nQit8XWMJvhfTt0rE=
client        DEBUG 	< Date: Thu, 08 Sep 2022 20:22:32 GMT
client        DEBUG 	< Via: 1.1 google
client        DEBUG 	< Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
protocol      DEBUG 	= connection is OPEN
protocol      DEBUG 	> TEXT '{"H": "Streaming", "M": "Subscribe", "A": [["He..."TimingData"]], "I": 0}' [294 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,1","S":1,"M":[]}' [47 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,2|e,1","G":"GfGc...Ddi9LKxqqBKQ==","M":[]}' [204 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,3|e,1|RF0,0","G"...f9HJCqRi9KFQw=","M":[]}' [230 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,4|e,1|RF0,0|RFz,...lP+GNX2/jnJg==","M":[]}' [280 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,5|e,1|RF0,0|RFz,...xpKJmDmaAon4I=","M":[]}' [304 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,6|e,1|RF0,0|RFz,...UL+hU9rpN0Vg==","M":[]}' [352 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,7|e,1|RF0,0|RFz,...qd3rJK/xLgiZ4=","M":[]}' [378 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,8|e,1|RF0,0|RFz,...ccUWypPSIOtA==","M":[]}' [427 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,9|e,1|RF0,0|RFz,...uXnDaf5KlPXpo=","M":[]}' [451 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,A|e,1|RF0,0|RFz,...PzZ1E5UI3HwA==","M":[]}' [499 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,B|e,1|RF0,0|RFz,...3o/z7tOStinIY=","M":[]}' [523 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,C|e,1|RF0,0|RFz,...Rq1K8n9E0t8g==","M":[]}' [571 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,D|e,1|RF0,0|RFz,...HcSgP4qKhp3vZg","M":[]}' [617 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,E|e,1|RF0,0|RFz,...JWub/3/zMqTw==","M":[]}' [646 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,F|e,1|RF0,0|RFz,...hJSOgWoJbzYtuY","M":[]}' [691 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,10|e,1|RF0,0|RFz...6f3+kbnmGdIA==","M":[]}' [720 bytes]
protocol      DEBUG 	< TEXT '{"C":"d-A221BE07-B,0|SbU,0|SbV,11|e,1|RF0,0|RFz...iLla3okwAOcGA=","M":[]}' [744 bytes]
protocol      DEBUG 	< TEXT '{"R":{"Heartbeat":{"Utc":"2022-09-04T15:20:16.0...hheld":false}},"I":"0"}' [80395 bytes]
protocol      DEBUG 	< TEXT '{}' [2 bytes]
protocol      DEBUG 	< TEXT '{}' [2 bytes]
protocol      DEBUG 	% sending keepalive ping
protocol      DEBUG 	> PING 97 9c 2b f1 [binary, 4 bytes]
protocol      DEBUG 	< PONG 97 9c 2b f1 [binary, 4 bytes]
protocol      DEBUG 	% received keepalive pong
protocol      DEBUG 	< TEXT '{}' [2 bytes]
```

### CODE REVIEW

**Comment by oscr** [comment]
If you are happy with the general approach I will cleanup the linting issues in the new module. Hopefully there is a way to skip the documentation step for the new module. As we talked about before: it's not a fastf1 feature we want to support so lets not present it as a feature :)

**Comment by theOehrly** [comment]
@oscr this looks good to me in general. I like that you included the reference to the original repository as well.

Regarding the documentation, the failure seems to be only caused by the example scripts. Sphinx-gallery will try to add all files in the examples directory to the examples gallery. It will only execute those files that start with `plot_` but it will still require other examples to be formatted correctly (doc strings, ...).
On that note, I think we want to add `example_fastf1_signalrclient.py` as an example to the documentation, yes. But I think we shouldn't add `example_signalr_connection.py` to the docs. It only serves as a reference for how to connect to other services, and we don't necessarily want to promote that as we don't want to support that, right? I'd keep the example file, probably in the examples folder as well. But I'd say it should be renamed, and we should define an ignore pattern, so that it is not added to the documentation.

The way that the documentation is set up, sphinx should not try and add the new submodule to the docs. We would need to explicitly add it, so that should be fine.

Apart from that, I'm not really happy with requiring the most current version of `requests` and `websockets`. In my opinion, that's an unnecessary constraint. `websockets` should use the same requirement as in the original `python-signalr-aio` module, except if that version is incompatible with python 3.10. 
We should either not add any version requirement for `requests` or duplicate the version requirement that `requests-cache` uses for `requests`. There's no reason why we would need to be stricter here.

**Comment by oscr** [comment]
@theOehrly Generally speaking I think more precise version constraints are a good thing. It's hard to keep a library compatible with older and future versions of a library. The more explicit you are, the less variants you need to test, support and less problems for users that happen to have an incompatible library version installed. With that said I don't know the Python ecosystem more than it's somewhat of a mess. So it's you call here.

**Question** what version constraints do you want? I picked latest and greatest to ensure it works with the (for Ubuntu) out of the box Python version. It might work with an older version. It might not. But I do know it works with the version I used at least. 

I will see if I can get sphinx to ignore the module and remove/fix the examples.

**Comment by oscr** [comment]
I added `.flake8` and `.gitattributes` to avoid having flake8 go crazy over the signalr module. Might be worth fixing when there are some nice tests for the module to ensure we don't fix it broken. :sweat_smile:

**Comment by theOehrly** [comment]
> I added `.flake8` and `.gitattributes` to avoid having flake8 go crazy over the signalr module. Might be worth fixing when there are some nice tests for the module to ensure we don't fix it broken. 😅

That's fine with me. It's a very sensible approach and we really don't need to fix the lint stuff there just because.

I've never use `.gitattributes`. Am I correct in assuming that the config you have added there is to circumvent the flake8 line lenght check on the diff?

**Comment by oscr** [comment]
> I've never use `.gitattributes`. Am I correct in assuming that the config you have added there is to circumvent the flake8 line length check on the diff?

This is exactly why it's there. flake8 seems to ignores the config file when the input is piped to it.

**Comment by oscr** [comment]
Thank you @theOehrly for the great review and research done on versions. Great work!

File: setup.cfg [hunk 1]
```diff
@@ -26,7 +26,8 @@ install_requires =
   thefuzz
   matplotlib>=3.3.3
   timple>=0.1.2
-  signalr-client-aio
+  requests>=2.28.0
```
Comment (theOehrly) [inline setup.cfg:29]: One minor thing here. `requests` is already defined above without a version constraint. We should define it twice. After that, this looks good to be merged

Comment (oscr) [reply setup.cfg:28]: @theOehrly No problem. Fixed.

**Comment by theOehrly** [comment]
You fixed it before I managed to correct the typo in my review comment... just how? 😂
Anyway, I'll merge this now. Thank you!

**Comment by oscr** [comment]
> You fixed it before I managed to correct the typo in my review comment... just how? joy Anyway, I'll merge this now. Thank you!

@theOehrly GitHub notifications alerted me as soon as the comment was made, and I actually thought about what you commented on before but had forgotten it. So I understood what you meant when I read it :grin:

**Pull request merged**